### PR TITLE
Refactoring part 5

### PR DESCRIPTION
**Pull request by IvanGoncharov**

### CODE REVIEW

**Comment by IvanGoncharov** [comment]
I think it's ready for merge.
@mohsen1 Can you please review it?

File: index.js [hunk 1]
```diff
@@ -66,65 +67,95 @@ var Converter = function() {};
...
+      }
+
+      this.convertApiDeclaration(declaration, operationTags);
+    });
+  }
+
+  return extend({},
+    this.aggregatePathComponents(resourceListing, apiDeclarations),
+    {
+      swagger: '2.0',
```
Comment (mohsen1) [inline index.js]: Are we removing extra keys here if original swagger does not have those features?

Comment (IvanGoncharov) [inline index.js]: Yes, it exactly what `extend` do, see [this](https://github.com/apigee-127/swagger-converter/blob/master/index.js#L646-L650).
Probably `extend({}, ...)` construction is confusing and I want create wrapper function for it.
Problem is I can't come up with proper name for it.
Do you have any suggestions?

**Comment by IvanGoncharov** [comment]
I remove some 'dead' code(`removeNonValues` function) in last commit.
It was left over from my experiment, sorry for inconvenience.

**Comment by mohsen1** [comment]
:+1: Thank you!

**Pull request merged**